aikau/src/main/resources/alfresco/lists/ItemSelectionMixin.js

```diff
@@ -0,0 +1,252 @@
+       * to "nodeRef" as the most likely use case is for working with Nodes but this can be configured
+       * to a different value.
+       *
+       * @instance
+       * @type {string}
+       * @default
+       */
+      itemKey: "nodeRef",
+
+      /**
+       * This indicates whether or not the mixing widget should subscribe to item selection events. This
+       * attribute is inspected by the [createItemSelectionSubscriptions]{@link module:alfresco/lists/ItemSelectionMixin#createItemSelectionSubscriptions}
+       * function to determine whether to bind the 
+       * [documentSelectionTopic]{@link module:alfresco/documentlibrary/_AlfDocumentListTopicMixin#documentSelectionTopic}
+       * to the [onItemSelection]{@link module:alfresco/lists/ItemSelectionMixin#onItemSelection} function.
+       * 
+       * @instance
+       * @type {boolean}
+       * @default
+       */
+      updateOnSelection: true,
+
+      /**
+       * This indicates whether or not clicking on the widget mixed by this module will trigger it's selection
+       * (it might be preferred to just update the CSS classes on selection rather than actually enabling
+       * item selection).
+       *
+       * @instance
+       * @type {boolean}
+       * @default
+       */
+      selectOnClick: true,
+
+      /**
+       * If [updateOnSelection]{@link module:alfresco/lists/ItemSelectionMixin#updateOnSelection} is configured to be
+       * true then this will setup the necessary subscriptions for item selection handling. Each time items are selected
+       * [onItemSelection]{@link module:alfresco/lists/ItemSelectionMixin#onItemSelection} will be called.
+       * 
+       * @instance
+       * @listens module:alfresco/documentlibrary/_AlfDocumentListTopicMixin#documentSelectionTopic
+       */
+      createItemSelectionSubscriptions: function alfresco_lists_ItemSelectionMixin__createItemSelectionSubscriptions() {
+         // Set up a subscription to handle file selection events, these will be along the lines of
+         // select all, select none, invert, etc. Each individual selector will respond to these
+         // events...
+         if (this.updateOnSelection)
+         {
+            this.alfSubscribe(this.documentSelectionTopic, lang.hitch(this, this.onItemSelection), this.getSelectionPublishGlobal(), this.getSelectionPublishToParent());
+         }
+      },
+
+      /**
+       * Returns whether or not selection subscriptions and publications should be made globally. By default
+       * this returns [publishGlobal]{@link module:alfresco/renderers/_PublishPayloadMixin#publishGlobal}.
+       *
+       * @instance
+       * @overridable
+       * @return {boolean} A boolean indicating whether or not to publish and subscribe to selection topics globally.
+       */
+      getSelectionPublishGlobal: function alfresco_lists_ItemSelectionMixin__getSelectionPublishGlobal() {
+         return this.publishGlobal;
```

Comment:
This variable does not exist, as far as I can tell, in the prototype chain. We need to either mixin another module or to include this in the defined variables.